homeassistant/components/androidtv/media_player.py

```diff
@@ -572,6 +572,11 @@ def adb_command(self, cmd):
             self.schedule_update_ha_state()
             return self._adb_response
 
+        if cmd == "LEARN_SENDEVENT":
+            self._adb_response = str(self.aftv.learn_sendevent())
+            self.schedule_update_ha_state()
+            return self._adb_response
+
```

Comment:
While I see this already in place, but it seem really odd to overload adb_shell commands with  random other strings. Seems it's totally different things. It blocks being able to send that string to the adb backend.
Seems those things really should be other services.

Reply:
The strings that it’s overloaded with aren’t legitimate ADB shell commands. That includes all of the "key" commands: HOME, LEFT, BACK, etc. For example, specifying the command "HOME" to the `androidtv.adb_command` service will send the command `input keyevent 3`. 

With "GET_PROPERTIES" and "LEARN_SENDEVENT" the `androidtv.adb_command` service sends complicated ADB shell commands, parses the results in the backend package, and returns those results to HA. The user could send those same shell commands and process the results themselves, but that’s too much trouble and people won’t do it. Therefore, I added these targets to the service. 

And if someone really did want to send the ADB shell command `LEARN_SENDEVENT` to their device (even though it will just respond with "command not found") they could just add a leading or trailing space.

Reply:
I think the commands have merit. Just don't think they should have been mixed into the same service. It would be breaking change to adjust now though.

Im okey with it, but may need a look by somebody else.

Reply:
I agree with Joakim. If the actual "LEARN_SENDEVENT" is not being send to ADB, it should not be there. Let's make it a new service

Reply:
Registering a new service is easy enough, but in my opinion this one piece of functionality isn't significant enough to deserve its own service.  I also don't want to have an excess of `androidtv.*` services.

Maybe there's a better approach?

As I mentioned above, commands like `"HOME"`, `"BACK"`, `"CENTER"`, etc. don't get sent verbatim via the `androidtv.adb_command` service.  Rather, they get translated into commands of the form `input keyevent x`.  Maybe these commands, along with this "LEARN_SENDEVENT" command, should be grouped together in a service.  Or maybe @elupus had it right initially when he suggested the remote platform -- it looks like Roku uses that -- although it does seem to me that the remote platform is a bit ambiguous.  I dunno.  

FYI, I'm almost done converting this integration to async.  After that, if someone is looking for an integration to modernize with config flow, maybe add a remote platform, and other fancy stuff, I think this would be a good candidate!

Reply:
It's fine to have multiple services. It's a lot easier to understand than having everything smashed into 1 service where each command can either be translated / trigger a bunch of commands or do something completely different.

Reply:
OK, I registered a separate `androidtv.learn_sendevent` service.